src/worker-client/index.ts

```diff
@@ -0,0 +1,202 @@
+import * as md5 from "md5";
+import PromiseWorker from "promise-worker";
+import EventAsyncIterator from "../event-async-iterator";
+import createCacheMetadata from "../helpers/create-cache-metadata";
+import EventTargetProxy from "../proxies/event-target";
+
+import {
+  ClientArgs,
+  ExportCachesResult,
+  ObjectMap,
+  PostMessageArgs,
+  PostMessageResult,
+  RequestOptions,
+  RequestResult,
+  RequestResultData,
+  ResponseCacheEntryResult,
+} from "../types";
+
+let instance: WorkerClient;
+
+export class WorkerClient {
+  public static async create(args: ClientArgs): Promise<WorkerClient> {
+    if (instance && isPlainObject(args) && !args.newInstance) return instance;
+
+    try {
+      const workerClient = new WorkerClient(args);
+      const webpackWorker = require("worker-loader?inline=true&fallback=false!../worker"); // tslint:disable-line
+      workerClient._createWorker(webpackWorker);
+      await workerClient._postMessage({ args, type: "create" });
+      instance = workerClient;
+      return instance;
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  private static _buildSubscriptionID(concatQuery: string, variables?: ObjectMap): string {
+    let id = concatQuery.replace(/\s/g, "");
+    if (variables) id += JSON.stringify(variables);
+    return md5(id);
+  }
+
+  private static _convertCacheabilityObjectMap({ cacheMetadata, ...otherProps }: PostMessageResult): RequestResultData {
+    return { cacheMetadata: createCacheMetadata({ cacheMetadata }), ...otherProps };
+  }
+
+  private static _isSubscription(concatQuery: string): boolean {
+    return concatQuery.startsWith("subscription");
+  }
+
+  private _eventEmitter: EventTargetProxy;
+  private _promiseWorker: PromiseWorker;
+  private _subscriptionsEnabled: boolean = false;
+  private _worker: Worker;
+
+  constructor(args: ClientArgs) {
+    if (args.subscriptions && isPlainObject(args.subscriptions)) {
+      this._eventEmitter = new EventTargetProxy();
+      this._subscriptionsEnabled = true;
+    }
+  }
+
+  public async clearCache(): Promise<void> {
+    try {
+      await this._postMessage({ type: "clearCache" });
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  public async exportCaches(tag: any): Promise<ExportCachesResult> {
+    try {
+      return this._postMessage({ tag, type: "exportCaches" });
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  public async getDataEntityCacheEntry(key: string): Promise<ObjectMap | undefined> {
+    try {
+      return this._postMessage({ key, type: "getDataPathCacheEntry" });
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  public async getDataEntityCacheSize(): Promise<number> {
+    try {
+      return this._postMessage({ type: "getDataEntityCacheSize" });
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  public async getDataPathCacheEntry(key: string): Promise<any> {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Update this function so that its implementation is not identical to the one on line 80. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS4144)